### PR TITLE
Regular expressions for names

### PR DESCRIPTION
**Pull request by dertseha**
## Why

Documentation specified that names "should" follow a pattern, yet it was unclear what the actual limits were.
Based on that, a more strict approach is now implemented, checking names against regular expressions.

## What

Both task and parameter names are checked against regular expressions. The expressions are similar, while I figured a task named `_` is feasible, having a parameter entered as `-_` is rather not. So, task names may start with an underscore, parameters not. Otherwise it's alphanumeric, plus dashes (in the middle).

## Checklist

- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated.
- [x] The code changes follow [Effective Go](https://golang.org/doc/effective_go).

### CODE REVIEW

**Comment by dertseha** [comment]
(Sorry to break your heart, dear Codecov...)

**Review by pellared — APPROVED** [review]
👍

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ Moreover, the parameters are set via CLI using the flag syntax.
 - Rename `Taskflow.MustRegister` method to `Register` and remove previous `Taskflow.Register` implemention.
 - Remove `Taskflow.Params` field and `TF.Params` method, add `Taskflow.Register*Param` methods and `Task.Params` field instead.
 - Remove `TF.Verbose`, add `Taskflow.VerboseParam` instead.
+- Enforce pattern for task and parameter names.
```
Comment (pellared) [inline CHANGELOG.md:35]: Describe the pattern

Comment (dertseha) [reply CHANGELOG.md]: done

**Pull request merged**